sqlglot/parser.py

```diff
@@ -6513,6 +6512,17 @@ def _parse_primary_key(
             and self._prev.token_type == TokenType.DESC
         )
 
+        # Parse and discard optional name for named primary keys (e.g., PRIMARY KEY pk_name (col))
+        if not in_props and self._match_set(
```

Comment:
i don't think this whole block is necessary, i think you can replace all of this code with self._parse_id_var right?